### DISCUSSION TITLE
Rotate inside group without shifting

### DISCUSSION THREAD

**Discussion by JuliDi** [cetz-package/cetz#246] [Q&A] [answered]
I would like to rotate the circle with the "+" inside by 45 degrees, but if I add the rotate command, it will also move. What am I doing wrong here? (A solution that can be factored out into a function to make it a reusable symbol would be greatly appreciated)

without rotate:
![Screenshot 2023-10-10 at 21 13 47](https://github.com/johannes-wolf/cetz/assets/20155974/e23a8262-dbef-4dea-a325-e56fd75bf289)

with rotate:
![Screenshot 2023-10-10 at 21 13 54](https://github.com/johannes-wolf/cetz/assets/20155974/44c298e1-a0f6-4876-b468-c3f270ff0875)

```typst
#import "@preview/cetz:0.1.2": canvas, draw, tree

#canvas(length: 1cm, {
  import draw: *
  line((0,0), (rel: (2, 0)))

  group({
    rotate(45deg)
    circle((rel:(0,0)), radius: 0.3, anchor: "left", name: "circle")
    line("circle.left", "circle.right")
    line("circle.top", "circle.bottom")
  }
  )
})
```

**Comment by johannes-wolf** [comment, marked as answer]
It is because the origin of the rotation is at (0,0) so you would have to translate everything first (you can use `set-origin(())` and then use `(0,0)` as your circle position). BUT: It is way more easy not using rotation at all, but other anchors:
```typst
#import "@preview/cetz:0.1.2": canvas, draw, tree

#canvas(length: 1cm, {
  import draw: *
  line((0,0), (rel: (2, 0)))

  group({
    circle((), radius: 0.3, anchor: "left", name: "circle")
    line("circle.top-left", "circle.bottom-right")
    line("circle.top-right", "circle.bottom-left")
  })
})
```

![grafik](https://github.com/johannes-wolf/cetz/assets/519002/ef60412f-4e41-4661-b40e-9c7b292fc054)

**Reply by JuliDi** [reply]
Thanks for your help. I think the anchors for `circle` are not documented yet, only for rect. Or do those anchor names work on all elements?

For anyone interested in the `set-origin` method, this works:
```typst
  group({
    set-origin((rel: (0.3, 0)))
    rotate(45deg)
    circle((0,0), radius: 0.3, anchor: "center", name: "circle")
    line("circle.left", "circle.right")
    line("circle.top", "circle.bottom")
  }
```

Edit: of course, the code @johannes-wolf suggested works a lot better.

**Reply by JuliDi** [reply]
One more issue that I ran into: When I use "line-right" anchor as the circle coordinate, the arrow tip will be inside the circle:
![Screenshot 2023-10-11 at 10 00 49](https://github.com/johannes-wolf/cetz/assets/20155974/106380ef-cb43-48c6-bb3b-a7270793ff8f)

```typst
#import "@preview/cetz:0.1.2": canvas, draw, tree

#canvas(length: 1cm, {
  import draw: *

  line((0,0), (rel: (2, 0)), mark: (end: ">"), name: "line")

  group({
    circle("line.right", radius: 0.3, anchor: "left", name: "circle")
    line("circle.top-left", "circle.bottom-right")
    line("circle.top-right", "circle.bottom-left")
  })
})
```

However if I use relative coordinates, e.g. `circle((rel: (0,0)), radius: 0.3, anchor: "left", name: "circle")`, it works as expected.
![Screenshot 2023-10-11 at 10 02 56](https://github.com/johannes-wolf/cetz/assets/20155974/56c9e0eb-3928-4847-90e5-f9480b059bd9)

This is especially bad if you want to place content above the line, then my only workaround is to place the content with relative coordinates and set `update: false` and then use relative coordinates again for the circle.

so this will again look bad:
```typst
  line((0,0), (rel: (2, 0)), mark: (end: ">"), name: "line")
  content("line", [$r_B (t)$], anchor: "bottom", padding: 0.2)
  
  group({
    circle("line.right", radius: 0.3, anchor: "left", name: "circle")
    line("circle.top-left", "circle.bottom-right")
    line("circle.top-right", "circle.bottom-left")
  })
```

and this, somewhat awkward coordinate definitions work:
```typst
  line((0,0), (rel: (2, 0)), mark: (end: ">"), name: "line")
  content((rel: (0,0), to: "line", update: false), [$r_B (t)$], anchor: "bottom", padding: 0.2)
  
  group({
    circle((rel: (0,0)), radius: 0.3, anchor: "left", name: "circle")
    line("circle.top-left", "circle.bottom-right")
    line("circle.top-right", "circle.bottom-left")
  })
 ```

**Reply by johannes-wolf** [reply]
Thank you, this has to to with the line getting shortened to have the arrow tip pointing directly to the coordinate. The anchor getting moved to is a bug! I will add a ticket.

**Reply by JuliDi** [reply]
Oh, I see. Thanks for taking care of it!

**Reply by johannes-wolf** [reply]
Ok. Sorry I had overseen you are using `line.right`, which is the wrong anchor to get a lines ending!
Use `"line.end"` and it works!

**Reply by JuliDi** [reply]
Using `"line.end"` it works as expected, thank you!
